src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -21,7 +23,7 @@
 class HeaderBag implements \IteratorAggregate, \Countable
 {
     protected $headers = array();
-    protected $cacheControl = array();
```

Comment:
ah ok. in that case perhaps ill add a new property then to hold the class and update the array as needed. and add a __clone for the new class